kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchBankInputFileType.java

```diff
@@ -0,0 +1,371 @@
+			bankRoutingNumber = KFSConstants.EMPTY_STRING;
+			bankName = KFSConstants.EMPTY_STRING;
+			message = KFSConstants.EMPTY_STRING;
+		}
+		
+		public BankReportLine(String bankRoutingNumber, String bankName, String message) {
+			super();
+			this.bankRoutingNumber = bankRoutingNumber;
+			this.bankName = bankName;
+			this.message = message;
+		}
+
+		public String getBankRoutingNumber() {
+			return bankRoutingNumber;
+		}
+
+		public void setBankRoutingNumber(String bankRoutingNumber) {
+			this.bankRoutingNumber = bankRoutingNumber;
+		}
+
+		public String getBankName() {
+			return bankName;
+		}
+
+		public void setBankName(String bankName) {
+			this.bankName = bankName;
+		}
+
+		public String getMessage() {
+			return message;
+		}
+
+		public void setMessage(String message) {
+			this.message = message;
+		}
+		
+	}
+	
+	@Override
+	public String getFileTypeIdentifer() {
+		return KFSConstants.ACHFileConstants.ACH_BANK_FILE_TYPE_IDENTIFIER;
+	}
+
+	@Override
+	public String getFileName(String principalName, Object parsedFileContents, String fileUserIdentifier) {
+		String fileName = "bank_" + principalName;
+		if (StringUtils.isNotBlank(fileUserIdentifier)) {
+			fileName += "_" + fileUserIdentifier;
+		}
+		fileName += "_" + dateTimeService.toDateTimeStringForFilename(dateTimeService.getCurrentDate());
+		
+		fileName = StringUtils.remove(fileName,  " ");
+		
+		return fileName;
+	}
+
+	@Override
+	public Object parse(byte[] fileByteContent) throws ParseException {
+		List<ACHBank> bankList = new ArrayList<ACHBank>();
+		List<BankReportLine> reportLines = new ArrayList<BankReportLine>();
+		String systemSource = KFSConstants.EMPTY_STRING;
+		ACHBank currentBank;
+		ACHBank existingBank;
+		BankReportLine reportLine;
+		Boolean saveBank = true;
+		BufferedReader bufferedFileReader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(fileByteContent)));
+		String fileLine;
+		
+		try {
+			while((fileLine = bufferedFileReader.readLine()) != null) {
+				LOG.info("Bank line: " + fileLine);
+				
+				//get a valid systemSource from the file
+				if (StringUtils.isBlank(systemSource)) {
+					systemSource = fileLine.substring(150, 152).trim();
+				}
+				
+				currentBank = new ACHBank(fileLine);
+				existingBank = getAchBank(currentBank.getBankRoutingNumber());
+				if (ObjectUtils.isNull(existingBank)) {
+					saveBank = validateBank(currentBank, reportLines);
+					if (saveBank) {
+						reportLine = new BankReportLine(currentBank.getBankRoutingNumber(), currentBank.getBankName(), KFSConstants.ACHFileConstants.ACH_BANK_LOADED_SUCCESSFULLY);
+						reportLines.add(reportLine);
+						businessObjectService.save(currentBank);
+						bankList.add(currentBank);
+					}
+				}
+			}
+			
+			writeReport(reportLines, systemSource);
+		}
+		catch (IOException e) {
+			LOG.error("Error encountered reading from file content", e);
+			throw new ParseException("Error encountered reading from file content", e);
+		}
```

Comment:
We need to watch that reader leaking as a leaking resource (I've seen a server brought down by this... I never forgot that lesson. I'd suggest:

`finally {

    IOUtils.closeQuietly(bufferedFileReader);

}`